src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -287,7 +288,118 @@ public static IResourceBuilder<ProjectResource> AddProject(this IDistributedAppl
+    /// <para>
+    /// This overload of the <see cref="AddCSharpApp(IDistributedApplicationBuilder, string, string)"/> method adds a C# project or file-based app to the application
+    /// model using a path to the file-based app .cs file, project file (.csproj), or project directory.
+    /// If the path is not an absolute path then it will be computed relative to the app host directory.
+    /// </para>
+    /// <example>
+    /// Add a file-based app to the app model via a file path.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// builder.AddCSharpApp("inventoryservice", @"..\InventoryService.cs");
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    /// </remarks>
+    [Experimental("ASPIRECSHARPAPPS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public static IResourceBuilder<ProjectResource> AddCSharpApp(this IDistributedApplicationBuilder builder, string name, string path)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
+        ArgumentNullException.ThrowIfNull(path);
+
+        return builder.AddCSharpApp(name, path, _ => { });
+    }
+
+    /// <summary>
+    /// Adds a C# project or file-based app to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used for service discovery when referenced in a dependency.</param>
+    /// <param name="path">The path to the file-based app file, project file, or project directory.</param>
+    /// <param name="configure">An optional action to configure the C# app resource options.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <para>
+    /// This overload of the <see cref="AddCSharpApp(IDistributedApplicationBuilder, string, string)"/> method adds a C# project or file-based app to the application
+    /// model using a path to the file-based app .cs file, project file (.csproj), or project directory.
+    /// If the path is not an absolute path then it will be computed relative to the app host directory.
+    /// </para>
+    /// <example>
+    /// Add a file-based app to the app model via a file path.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// builder.AddCSharpApp("inventoryservice", @"..\InventoryService.cs", o => o.LaunchProfileName = "https");
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    /// </remarks>
+    [Experimental("ASPIRECSHARPAPPS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public static IResourceBuilder<CSharpAppResource> AddCSharpApp(this IDistributedApplicationBuilder builder, [ResourceName] string name, string path, Action<ProjectResourceOptions> configure)
```

Comment:
Does the resource have the new C# icon?

What kind of resource is this? If it's considered a project resource then you could find the place that checks for the project extension and add `.cs` to the `.csproj` check (this is what we do for projects currently)

If it is an executable resource then you could add WithResourceName and specify the C# icon (this is what we do for NodeJS and Python executable resources)

Reply:
See https://github.com/dotnet/aspire/pull/11502/files#diff-9280d171e1a391b74a79aae80d47a9e92a8673acffca1e4edba00f4c3e7a097bR54

Reply:
<img width="1904" height="665" alt="image" src="https://github.com/user-attachments/assets/82601d12-2f2a-42c8-a74a-70c60c510c1d" />